File: 13-roman-to-integer/roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        values = {
            "I": 1,
            "V": 5,
            "X": 10,
            "L": 50,
            "C": 100,
            "D": 500,
            "M": 1000  
        }

        total = 0
        prevValue = 0

        for char in s:
            currValue = values[char]

            if currValue > prevValue:
                total -= prevValue
                total += (currValue - prevValue)
            else:
                total += currValue
            
            prevValue = currValue
        return total

File: 13-roman-to-integer/roman-to-integer.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function(s) {
    const values = {
        "I": 1,
        "V": 5,
        "X": 10,
        "L": 50,
        "C": 100,
        "D": 500,
        "M": 1000
    }

    let sum = 0;
    let prevValue = 0;

    for (let i = 0; i < s.length; i ++) {
        currValue = values[s[i]];
        if (currValue > prevValue) {
            sum -= prevValue;
            sum += (currValue - prevValue);
        } else {
            sum += currValue;
        };

        prevValue = currValue;
    };

    return sum
};